test/tools/yulopti.cpp

```diff
@@ -278,7 +301,10 @@ Allowed options)",
 	string input;
 	try
 	{
-		input = readFileAsString(arguments["input-file"].as<string>());
+		if (arguments.count("input-file"))
+			input = readFileAsString(arguments["input-file"].as<string>());
+		else if (nonInteractive)
+			input = readUntilEnd(cin);
```

Comment:
I think it would be better to require `-` to use stdin. That would be more consistent with how `solc` works.

I'm fine with it staying like this in this PR in the interest of getting it through faster though :) Could be changed separately if you agree it's a good idea.

Reply:
Yeah, that's indeed better, not sure why it didn't cross my mind :-), I'll do it here.